rich/spinner.py

```diff
@@ -79,7 +79,8 @@ def render(self, time: float) -> "RenderableType":
             self.speed = self._update_speed
             self._update_speed = 0.0
 
-        if not self.text:
+        # This normally can't be str, unless someone assigned it later.
+        if not self.text:  # type: ignore[truthy-bool]
```

Comment:
Typewise this can't happen, but I could see it possibly happing at runtime, so just added an ignore and note here.

Reply:
Yes, that appears to be it, mypy assigns `self.text: RenderableType` because Text fits that Protocol. But the `__bool__` behavior is unique to Text, so this confuses it below. Making this explicit solves the issue.